app/models/derived_variables/case_log_variables.rb

```diff
@@ -0,0 +1,168 @@
+module DerivedVariables::CaseLogVariables
+  RENT_TYPE_MAPPING = { 0 => 1, 1 => 2, 2 => 2, 3 => 3, 4 => 3, 5 => 3 }.freeze
+
+  def set_derived_fields!
+    # TODO: Remove once we support supported housing logs
+    self.needstype = 1 unless needstype
+    if rsnvac.present?
+      self.newprop = has_first_let_vacancy_reason? ? 1 : 2
+    end
+    self.incref = 1 if net_income_refused?
+    self.renttype = RENT_TYPE_MAPPING[rent_type]
+    self.lettype = get_lettype
+    self.totchild = get_totchild
+    self.totelder = get_totelder
+    self.totadult = get_totadult
+    self.refused = get_refused
+    self.ethnic = 17 if ethnic_refused?
+    if %i[brent scharge pscharge supcharg].any? { |f| public_send(f).present? }
+      self.brent ||= 0
+      self.scharge ||= 0
+      self.pscharge ||= 0
+      self.supcharg ||= 0
+      self.tcharge = brent.to_f + scharge.to_f + pscharge.to_f + supcharg.to_f
+    end
+    if period.present?
+      self.wrent = weekly_value(brent) if brent.present?
+      self.wscharge = weekly_value(scharge) if scharge.present?
+      self.wpschrge = weekly_value(pscharge) if pscharge.present?
+      self.wsupchrg = weekly_value(supcharg) if supcharg.present?
+      self.wtcharge = weekly_value(tcharge) if tcharge.present?
+      if is_supported_housing? && chcharge.present?
+        self.wchchrg = weekly_value(chcharge)
+      end
+    end
+    self.has_benefits = get_has_benefits
+    self.tshortfall_known = 0 if tshortfall
+    self.wtshortfall = if tshortfall && receives_housing_related_benefits?
+                         weekly_value(tshortfall)
+                       end
+    self.nocharge = household_charge&.zero? ? 1 : 0
+    self.housingneeds = get_housingneeds
+    if is_renewal?
+      self.underoccupation_benefitcap = 2 if collection_start_year == 2021
+      self.homeless = 1
+      self.referral = 0
+      self.waityear = 2
```

Comment:
For `homeless` and `waityear` we will have to update values in production.
The former cannot have value 2, the latter cannot have value 1 for 22/23 forms.

Reply:
Do we know how many logs this affects?

Reply:
`homeless` value 2 was present on a single record last time I checked.
I haven't checked `waityear` yet.

Reply:
1 record (renewal = 1 and homeless = 2)
1 record (renewal = 1 and waityear = 1)

Same record ID

Reply:
Meaning that 1 record will become invalid? I guess we should just clear the value and have it set back to in progress and inform them?

Reply:
we just need to change the values for this single record